packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.tsx

```diff
@@ -89,14 +53,19 @@ function BlogPostPageContent(props: Props): JSX.Element {
 }
 
 export default function BlogPostPage(props: Props): JSX.Element {
+  const BlogPostContent = props.content;
   return (
-    <HtmlClassNameProvider
-      className={clsx(
-        ThemeClassNames.wrapper.blogPages,
-        ThemeClassNames.page.blogPostPage,
-      )}>
-      <BlogPostPageMetadata {...props} />
-      <BlogPostPageContent {...props} />
-    </HtmlClassNameProvider>
+    <BlogPostProvider content={props.content} isBlogPostPage>
```

Comment:
wonder if the provider shouldn't be applied directly in BlogPostItem 🤷‍♂️ (as we apply providers in BlogPostItemList)

At the same time, it's convenient to be able to call `useBlogPost()` in page metadata that remains upper in the tree than BlogPostItem

Maybe we should stick to using props in the `XyzPage` components, and only use context in `XyzItem` ? (like we do in other places)

Reply:
> Maybe we should stick to using props in the `XyzPage` components, and only use context in `XyzItem`? (like we do in other places)

I'm fine with that, but blog doesn't have nested routes like docs do.

Reply:
> I'm fine with that, but blog doesn't have nested routes like docs do.

Actually want to move later to a similar setup

The blog sidebar should become more powerful over time (ie custom items etc), its data should be available in context easily, and the sidebar should avoid unnecessary unmount/remount on blog navigation (custom stateful components lose state)